msmart/device/AC/command.py

```diff
@@ -114,9 +114,23 @@ def payload(self) -> bytes:
         beep = 0x40 if self.beep_on else 0
         power = 0x1 if self.power_on else 0
 
-        # Build target temp and mode byte
-        fractional, integral = math.modf(self.target_temperature)
-        temperature = (int(integral) & 0xF) | (0x10 if fractional > 0 else 0)
+        # Get integer and fraction components of target temp
+        fractional_temp, integral_temp = math.modf(self.target_temperature)
+        integral_temp = int(integral_temp)
+
+        if 17 <= integral_temp <= 30:
+            # Use primary method
+            temperature = (integral_temp - 16) & 0xF
+            temperature_alt = 0
+        else:
+            # Out of range, use alternate method
+            # TODO additional range possible according to Lua code
+            temperature = 0
+            temperature_alt = (integral_temp - 12) & 0x1F

```

Comment:
This block feels like it should be a ternary statement but I've been unhappy with every combination I tried